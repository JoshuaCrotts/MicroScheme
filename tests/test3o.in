//
// Power and bitwise operator testing
//

void printlnd(int i) {
	printd(i);
	printc('\n');
}

void main() {
	int i;
	int j;
	int k;
	
	i = 2 ** 3 ** 2; // 512
	printlnd(i);
	i = i >> 9; 
	printlnd(i);
	i++;
	j = (i << 4 & 0x67) ** 3;
	printlnd(j / (i<<1));
	printlnd(j);
	k = (j << i);
	printlnd(~(-k ^ 0xffff));
}
