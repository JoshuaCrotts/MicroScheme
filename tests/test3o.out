512
1
8192
32768
65536
